example/bwmf/proto/bwmf.proto

```diff
@@ -15,19 +15,14 @@ message Request {
 
 message Response {
   uint64 blockId = 1;
-  DenseMatrixShard shard = 2;
+  MatrixShard shard = 2;
 }
 
-message DenseMatrixShard {
-  message DenseRow {
-    repeated float at = 1;
+message MatrixShard {
+  message Row {
+    uint32 rowId = 1;
+    repeated uint32 colIds = 2;
+    repeated float values = 3;
   }
-  repeated DenseRow row = 1;
-}
-
-message SparseMatrixShard {
-  message SparseRow {
-    map<int32, float> at = 1;
-  }
-  repeated SparseRow row = 1;
+  repeated Row rows = 1;
 }
```

Comment:
Agreed. That will be convenient to work with linear algebra libs.

BTW, if my understanding is right, sparse should be a bool field rathar than a repeated bool. Is it?

Also, in BWMF, we access data from multiple matrices where sequential indices for one matrix might be random for another. This will make accessing inefficient. In PR #83 , I use the original sparse map-based representation to accelerate random accessing.